src/block.rs

```diff
@@ -14,35 +19,106 @@ pub struct Block {
 
 impl Block {
     #[rustfmt::skip]
-    pub fn encode(&self) -> Bytes {
+    pub fn encode(&self, c: Option<CompressionCodec>) -> Bytes {
```

Comment:
I think @rodesai was trying to avoid this by compressing the block bytes in the SsTableFormat.

https://github.com/slatedb/slatedb/issues/10#issuecomment-2291882147